tests/search/test_search.py

```diff
@@ -1,106 +1,140 @@
-#             status="active",
-#             funding_goal=10000,
-#         )
-#
-#     @mock.patch("search.views.StartupDocument.search")
-#     def test_startup_search_mocked(self, mock_search):
-#         """Test startup search with mocked Elasticsearch"""
-#
-#         # Prepare mock search result
-#         mock_execute = mock.Mock()
-#         mock_execute.__iter__ = lambda s: iter([mock.Mock(id=self.startup.id)])
-#         mock_search.return_value.execute.return_value = mock_execute
-#
-#         # Call API
-#         url = reverse("startup-search")
-#         response = self.client.get(url, {"q": "AI"})
-#
-#         # Assertions
-#         self.assertEqual(response.status_code, 200)
-#         self.assertEqual(len(response.data), 1)
-#         self.assertEqual(response.data[0]["company_name"], "Test Startup")
-#
-#     @mock.patch("search.views.ProjectDocument.search")
-#     def test_project_search_mocked(self, mock_search):
-#         """Test project search with mocked Elasticsearch"""
-#
-#         # Prepare mock search result
-#         mock_execute = mock.Mock()
-#         mock_execute.__iter__ = lambda s: iter([mock.Mock(id=self.project.id)])
-#         mock_search.return_value.execute.return_value = mock_execute
-#
-#         # Call API
-#         url = reverse("project-search")
-#         response = self.client.get(url, {"q": "AI"})
-#
-#         # Assertions
-#         self.assertEqual(response.status_code, 200)
-#         self.assertEqual(len(response.data), 1)
-#         self.assertEqual(response.data[0]["title"], "Health AI")
-#
-#     def test_startup_search_empty_query(self):
-#         """Test empty query for startups should return empty list"""
-#         url = reverse("startup-search")
-#         response = self.client.get(url, {"q": ""})
-#         self.assertEqual(response.status_code, 200)
-#         self.assertEqual(len(response.data), 0)
-#
-#     def test_project_search_empty_query(self):
-#         """Test empty query for projects should return empty list"""
-#         url = reverse("project-search")
-#         response = self.client.get(url, {"q": ""})
-#         self.assertEqual(response.status_code, 200)
-#         self.assertEqual(len(response.data), 0)
+from decimal import Decimal
+from unittest import mock
 
+from django.urls import reverse
+from rest_framework.test import APITestCase
 
+from users.models import User, UserRole
+from startups.models import Startup, Industry, Location
+from projects.models import Project, Category
 
+import search.views as sv
 
 
+class SearchTests(APITestCase):
+    def setUp(self):
+        # Role / user
+        role_user, _ = UserRole.objects.get_or_create(role=UserRole.Role.USER)
+        self.user = User.objects.create_user(
+            email="test@example.com",
+            password="password123",
+            role=role_user,
+            is_active=True,
+        )
+        self.client.force_authenticate(user=self.user)
```

Comment:
Forcing authentication with a user at setup may cause issues if tests want unauthenticated states. Only force authentication in individual test methods as needed.